deployer/src/ec2/utils.rs

```diff
@@ -0,0 +1,142 @@
+//! Utility functions for interacting with EC2 instances
+
+use crate::ec2::Error;
+use tokio::process::Command;
+use tokio::time::{sleep, Duration};
+use tracing::warn;
+
+/// Maximum number of SSH connection attempts before failing
+pub const MAX_SSH_ATTEMPTS: usize = 30;
+
+/// Maximum number of polling attempts for service status
+pub const MAX_POLL_ATTEMPTS: usize = 30;
+
+/// Interval between retries
+pub const RETRY_INTERVAL: Duration = Duration::from_secs(10);
+
+/// Fetch the public IPv4 address of a machine
+pub async fn get_public_ip() -> Result<String, Error> {
+    let result = reqwest::get("https://ipv4.icanhazip.com")
```

Comment:
Lol

I guess AWS has some other methods too if you want to use

https://stackoverflow.com/questions/38679346/get-public-ip-address-on-current-ec2-instance

Reply:
This is only used for the "deployer IP". We use AWS for all ec2 IPs 👍 .

Reply:
Turns out this is actually run by Cloudflare, so just opted to add a comment: https://major.io/p/a-new-future-for-icanhazip/